### PR TITLE
Ocular rewrite

### LINKED ISSUE #4938: Add an option to stop my-ocular from displaying your Ocular status on your own profile/posts
### Is your request related to a problem? Please describe.
No. Today I was on my profile and realized that my Ocular status was displaying (with `my-ocular` enabled, of course). That's what's intended, and it's fine, but it clutters the display with information that I don't actually need or want.

Also, when I'm on the forums, I see my Ocular status. That's less intrusive, but still related.

### Describe the solution you'd like
I'd like it if there was an option in `my-ocular` to hide your Ocular status on your profile and/or forum posts. This would help to declutter the display.

### Describe alternatives you've considered
Deal with it

### Additional context
None

### PR DESCRIPTION
**Pull request by Norbiros**
Fixes https://github.com/ScratchAddons/ScratchAddons/pull/4739#issuecomment-1204786526
Resolves #4938

### Changes
- I re-wrote code for ocular
- I fixed some small bugs (for example with user id addon)
- I added new setting #4938

### To Do
Add comments and change some things in `reactions.js`

## Additional Info
@mxmou can you check code in `discuss.js` (I am talking about dynamic enable/disable function)?

### Tests

Everything works on Firefox.

### CODE REVIEW

File: addons/my-ocular/addon.json [hunk 1]
```diff
@@ -1,15 +1,16 @@
 {
-  "name": "ocular integration",
+  "name": "Ocular integration",
```
Comment (CST1229) [inline addons/my-ocular/addon.json:2]: `ocular` (all lowercase) is the official name, I'm pretty sure it shouldn't be capitalized.
```suggestion
  "name": "ocular integration",
```

Comment (Norbiros) [reply addons/my-ocular/addon.json:2]: But it looks better? @jeffalo what do you think?

Comment (jeffalo) [reply addons/my-ocular/addon.json:2]: ocular is spelt ocular.

File: addons/my-ocular/addon.json [hunk 2]
```diff
@@ -60,6 +61,22 @@
       "type": "boolean",
       "default": true
     },
+    {
+      "name": "Show statuses ",
```
Comment (apple502j) [inline addons/my-ocular/addon.json:65]: Names should not have trailing space.

File: addons/my-ocular/addon.json [hunk 3]
```diff
@@ -60,6 +61,22 @@
...
+      "id": "show-status",
+      "type": "select",
+      "potentialValues": [
+        {
+          "id": "everywhere",
+          "name": "everywhere"
+        },
+        {
+          "id": "others",
+          "name": "only on other's profiles"
```
Comment (apple502j) [inline addons/my-ocular/addon.json:75]:
```suggestion
          "name": "only on others' profiles"
```

File: addons/my-ocular/addon.json [hunk 4]
```diff
@@ -69,5 +86,10 @@
   ],
   "versionAdded": "1.4.0",
   "tags": ["forums", "profiles", "community"],
-  "enabledByDefault": false
+  "enabledByDefault": false,
+  "latestUpdate": {
+    "version": "1.29.0",
+    "temporaryNotice": "Code re-write and new setting!",
```
Comment (apple502j) [inline addons/my-ocular/addon.json:92]: Do not mention code rewrite.

**Review by apple502j — CHANGES_REQUESTED** [review]

**Review by apple502j — APPROVED** [review]

**Review by jeffalo — APPROVED** [review]
lgtm.

File: addons/my-ocular/addon.json [hunk 5]
```diff
@@ -31,8 +32,8 @@
   ],
   "userstyles": [
     {
-      "url": "discuss.css",
-      "matches": ["https://scratch.mit.edu/discuss/topic/*"],
+      "url": "status-dot.css",
+      "matches": ["topics", "https://scratch.mit.edu/users/*/"],
       "if": {
         "settings": { "discuss": true }
       }
```
Comment (WorldLanguages) [inline addons/my-ocular/addon.json:39]: Disabling CSS in profiles if `SHOW STATUSES ON FORUMS` is disabled, but `SHOW STATUSES ON PROFILES` is enabled? Looks like no one noticed this!

**Pull request merged**